docs/specification/2025-03-26/basic/authorization.mdx

```diff
@@ -8,48 +8,141 @@ title: Authorization
+   Server Metadata ([RFC 8414](https://datatracker.ietf.org/doc/html/rfc8414)).
+
+4. MCP servers **MUST** implement and support Auth 2.0 Protected Resource Metadata
+   ([RFC DRAFT](https://datatracker.ietf.org/doc/draft-ietf-oauth-resource-metadata/13/)).
+
+### 2.2 Client OAuth 2.1 Authorization
+
+When authorization is required and not yet proven by the client, servers **MUST** respond with
+`HTTP 401 Unauthorized`.
+
+A `HTTP 401 Unauthorized` response **MUST** include a `WWW-Authenticate` header with
+the URL to the protected resource metadata document hosted by the MCP Server in
+the following format:
+
+```http
+WWW-Authenticate: Bearer resource_metadata="https://resource.example.com/.well-known/oauth-protected-resource",
+     scope="mcp:read mcp:write"
+```
+
+A MCP server **MUST** host its Protected Resource Metadata (PRM) document, as outlined in
+[Section 3 of the RFC draft](https://www.ietf.org/archive/id/draft-ietf-oauth-resource-metadata-13.html#section-3).
+
+A MCP server **MUST** return the minimum required set of scopes in the `scope` parameter within the
+`WWW-Authenticate` header in the initial handshake as well as any additional scopes in subsequent responses
+that require incremental authorization.
+
+A MCP server **MUST** include its AS configuration in the PRM document as outlined in
+[Section 2 of the RFC draft](https://www.ietf.org/archive/id/draft-ietf-oauth-resource-metadata-13.html#section-2).
+
+A MCP client **MUST** issue a `GET` request to the URL in `resource_metadata` to get information about
+the AS.
 
-2. MCP auth implementations **SHOULD** support the OAuth 2.0 Dynamic Client Registration
-   Protocol ([RFC7591](https://datatracker.ietf.org/doc/html/rfc7591)).
+A MCP client **MUST** verify that the `resource` value in the PRM document matches the MCP server
+protocol and fully-qualified domain name (FQDN).
 
-3. MCP servers **SHOULD** and MCP clients **MUST** implement OAuth 2.0 Authorization
-   Server Metadata ([RFC8414](https://datatracker.ietf.org/doc/html/rfc8414)). Servers
-   that do not support Authorization Server Metadata **MUST** follow the default URI
-   schema.
+A MCP client **SHOULD** evaluate or permit the user to evaluate the trustworthiness of AS
+discovered through the PRM document to prevent phishing. For example, an MCP client could prompt 
+the user "Do you want to login with `https://authserver.example`?"
 
-### 2.1.1 OAuth Grant Types
+A MCP client **MUST** use `authorization_servers` values from the PRM document to construct
+the OAuth 2.0 metadata discovery document path, following
+[RFC 8414](https://datatracker.ietf.org/doc/html/rfc8414#section-3).
+
+A MCP client **MAY** use the AS-provided registration endpoint to register itself with the AS.
+
+A MCP client **MAY** be pre-registered with an AS and not require dynamic registration.
+
+A MCP client **MUST** send the `resource` identifier to the AS, following
```

Comment:
Is there an implicit requirement that the AS should support and implement RFC 8707? There is another point in section 2.4.4 which also implies that. If so, this could be difficult to enforce given that the AS could audience-restrict the token using a different strategy and the outcome would be the same.